src/silly_sprites/event/base_handler.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "event_concept.hpp"
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+
+namespace sly::event {
+    template<Event E>
+    class Handler {
+    private:
+        using handler_ty = Handler<E>;
+
+    public:
+        Handler() = default;
+        Handler(handler_ty const& other) {
+            E::copy(&other, this);
+        }
+        Handler(handler_ty&& other) noexcept {
+            E::move(&other, this);
+        }
+        handler_ty& operator=(handler_ty const& other) {
+            if (this == &other) {
+                return *this;
+            }
+            E::copy(&other, this);
+            return *this;
+        }
+        handler_ty& operator=(handler_ty&& other) {
+            if (this == &other) {
+                return *this;
+            }
+            E::move(&other, this);
+            return *this;
+        }
+
+        virtual ~Handler() {
+            E::disconnect(this);
+        }
+
+        virtual void on_event(E const& event) = 0;
+    };
+
+    template<Event E>
+    struct Base {
+    private:
+        using handler_ty = Handler<E>*;
+        using handler_const_ty = Handler<E> const*;
+        inline static std::vector<handler_ty> m_handlers;
+
```

Comment:
Again, type names in snake_case instead of UpperCase.

Also: The prefix `m_` should only be used for non-static data members. I think the styleguide says to prefix static data members with `s_`.